### PR TITLE
Action Points reference number moved to db

### PR DESCRIPTION
**Pull request by reinbach**
[ch12761]

See https://github.com/unicef/etools/pull/2412

### CODE REVIEW

File: src/etools/applications/action_points/signals.py [hunk 1]
```diff
@@ -20,5 +19,6 @@ def action_point_updated_receiver(instance, created, **kwargs):
     if relevant_ap_instance:
         if created:
             instance.send_email(instance.assigned_to, email_template, cc=[instance.assigned_by.email])
-        elif instance.tracker.has_changed('assigned_to'):
-            instance.send_email(instance.assigned_to, email_template)
+        elif not instance.tracker.has_changed('reference_number'):
```
Comment (domdinicola) [inline src/etools/applications/action_points/signals.py]: @reinbach why are we adding this?

File: src/etools/applications/action_points/models.py [hunk 2]
```diff
@@ -195,7 +204,7 @@ def get_mail_context(self, user=None):
         return {
             'person_responsible': self.assigned_to.get_full_name(),
             'assigned_by': self.assigned_by.get_full_name(),
-            'reference_number': self.reference_number,
+            'reference_number': self.get_reference_number(),
```
Comment (domdinicola) [inline src/etools/applications/action_points/models.py]: why we can't we use the reference_number? Is because the email can be send before the creation?

Comment (reinbach) [reply src/etools/applications/action_points/models.py]: Yes, an email is sent out on `created` signal. Reference number only set in subsequent save.

File: src/etools/applications/action_points/signals.py (hunk 1 above)
Comment (reinbach) [reply src/etools/applications/action_points/signals.py]: This is to ignore the subsequent save that happens when setting the reference number, on this save just the reference number is changed. So do not want to accidentally send out `assign_to` email

Comment (robertavram) [reply src/etools/applications/action_points/signals.py]: hmm.. Maybe I'm wrong but it looks to me that the purpose of l23 originally was to send another email on a subsequent save only if the "assgined_to" was changed.. so if 20 saves in.. someone changes the pereson responsible for the action point, that person will recieve the email.. I think that logic will break with this proposed change.

File: src/etools/applications/action_points/models.py (hunk 2 above)
Comment (robertavram) [reply src/etools/applications/action_points/models.py]: on `get_reference_number` method can we return current ref number if exists on record if not the new number generated.. that way we can rest assured that there's no funny business happening.. especially if we change the logic for gen ref numbers in the future.

File: src/etools/applications/action_points/signals.py (hunk 1 above)
Comment (reinbach) [reply src/etools/applications/action_points/signals.py]: If we don't have this logic the `ActionPointsEmailsTestCase.test_create_unlinked` test results in 3 emails being sent out.

File: src/etools/applications/action_points/models.py (hunk 2 above)
Comment (reinbach) [reply src/etools/applications/action_points/models.py]: done

**Pull request merged**